packages/contracts/src/dollar/libraries/LibStaking.sol

```diff
@@ -1,532 +1,514 @@
+        stakingStore.rewardToken = IERC20Ubiquity(newRewardToken);
     }
 
     /**
-     * @notice Mints a staking share on deposit
-     * @param to Address where to mint a staking share
-     * @param lpAmount Amount of LP tokens
-     * @param shares Amount of shares
-     * @param endBlock Staking share end block
-     * @return Staking share id
+     * @notice Sets start block when staking should be active
+     * @param newStartBlock Block number when staking should be active
      */
-    function _mint(
-        address to,
-        uint256 lpAmount,
-        uint256 shares,
-        uint256 endBlock
-    ) internal returns (uint256) {
-        uint256 _currentShareValue = currentShareValue();
-        require(
-            _currentShareValue != 0,
-            "Staking: share value should not be null"
-        );
-        // set the lp rewards debts so that this staking share only get lp rewards from this day
-        uint256 lpRewardDebt = (shares * stakingStorage().accLpRewardPerShare) /
-            1e12;
-        return
-            StakingShare(LibAppStorage.appStorage().stakingShareAddress).mint(
-                to,
-                lpAmount,
-                lpRewardDebt,
-                endBlock
-            );
+    function setStakingStartBlock(uint256 newStartBlock) internal {
+        require(newStartBlock >= block.number, "Can't start in the past");
+        StakingStorage storage stakingStore = stakingStorage();
+        stakingStore.startBlock = newStartBlock;
     }
 
     /**
-     * @notice Returns staking share info
-     * @param _id Staking share id
-     * @return bs Array of amount of shares and reward debt
-     * @return stake Stake info
+     * @notice Updates the given pool's Governance token allocation points
+     * @param poolId Pool id
+     * @param allocationPoints New allocation points
+     * @param withUpdate Whether to trigger update on all staking pools
      */
-    function _checkForLiquidity(
-        uint256 _id
-    )
-        internal
-        view
-        returns (uint256[2] memory bs, StakingShare.Stake memory stake)
-    {
-        address stakingAddress = LibAppStorage.appStorage().stakingShareAddress;
-        require(
-            IERC1155Ubiquity(stakingAddress).balanceOf(msg.sender, _id) == 1,
-            "Staking: caller is not owner"
-        );
-        StakingShare staking = StakingShare(stakingAddress);
-        stake = staking.getStake(_id);
-        require(
-            block.number > stake.endBlock,
-            "Staking: Redeem not allowed before staking time"
-        );
+    function updateStakingPool(
+        uint256 poolId,
+        uint256 allocationPoints,
+        bool withUpdate
+    ) internal {
+        StakingStorage storage stakingStore = stakingStorage();
+
+        require(poolId < stakingStore.poolInfo.length, "Pool does not exist");
+
+        if (withUpdate) {
+            massUpdateStakingPools();
```

Comment:
I would add a `poolLength` as a parameter instead of a `bool` so that if `massUpdateStakingpools` reverts due to all gas being consumed we can still call it with a smaller poolLength

Reply:
Refactored to use individual pool ids for mass update in https://github.com/ubiquity/ubiquity-dollar/pull/987/commits/c57b8f0ad9e2f380cdb9d7378f54424f3cd2e0a9